release/docker/build-image-multi-arch.sh

```diff
@@ -0,0 +1,141 @@
+    docker buildx use default
+    docker buildx rm $BUILDER_NAME > /dev/null 2>&1
+}
+
+function cleanup_all() {
+    cleanup_docker_buildx
+    File_Delete $DIR
+}
+while getopts ":ho:v:f:p:a:r:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        o)
+            TARBALL=`realpath $OPTARG`
+            ;;
+        v)
+            VERSION=$OPTARG
+            ;;
+        f)
+            DOCKERFILE=$OPTARG
+            ;;
+        p)
+            PRODUCT=$OPTARG
+            ;;
+        a)
+            ARCHITECTURE=$OPTARG
+            ;;
+        r)
+            REPOSITORY=$OPTARG
+            ;;
+        :)
+            echo "-${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+# Validate the required parameters to present
+if [ -z "$VERSION" ] || [ -z "$DOCKERFILE" ] || [ -z "$PRODUCT" ] || [ -z "$ARCHITECTURE" ]; then
+  echo "You must specify '-v VERSION', '-f DOCKERFILE', '-p PRODUCT', '-a ARCHITECTURE'"
+  usage
+  exit 1
+else
+  echo $VERSION $DOCKERFILE $PRODUCT $ARCHITECTURE
+  IFS=', ' read -r -a ARCHITECTURE_ARRAY <<< "$ARCHITECTURE"
+  IFS=', ' read -r -a TARBALL_ARRAY <<< "$TARBALL"
+fi
+
+# Warning docker desktop
+if (! docker buildx version)
+then
+    echo -e "\n* You MUST have Docker Desktop to use buildx for multi-arch images."
+    exit 1
+fi
+
+# Prepare docker buildx
+trap cleanup_all TERM INT EXIT
+DIR=`Temp_Folder_Create`
+echo New workspace $DIR
+echo -e "\n* Prepare docker buildx"
+docker buildx use default
+docker buildx create --name $BUILDER_NAME --use
+docker buildx inspect --bootstrap
+
+# Check buildx status
+echo -e "\n* Check buildx status"
+docker buildx ls | grep $BUILDER_NAME
+docker ps | grep $BUILDER_NAME
+
+
+# Copy configs
+cp -v config/${PRODUCT}/* $DIR/
+cp -v ../../scripts/opensearch-onetime-setup.sh $DIR/
+
+# Copy TGZ
+if [ -z "$TARBALL" ]; then
+    for index in ${!ARCHITECTURE_ARRAY[@]}
+    do
+        # No tarball file specified so download one
+        arch_uname=`echo ${ARCHITECTURE_ARRAY[$index]} | sed 's/x64/x86_64/g;s/arm64/aarch64/g'`
+        URL="https://artifacts.opensearch.org/releases/bundle/${PRODUCT}/${VERSION}/${PRODUCT}-${VERSION}-linux-${ARCHITECTURE_ARRAY[$index]}.tar.gz"
+        echo -e "\nDownloading ${PRODUCT} arch ${ARCHITECTURE_ARRAY[$index]} version ${VERSION} from ${URL}"
+        curl -f $URL -o ${DIR}/${PRODUCT}-${arch_uname}.tgz || exit 1
+    done
+    ls -l $DIR
+else
+    for index in ${!ARCHITECTURE_ARRAY[@]}
+    do
+        arch_uname=`echo ${ARCHITECTURE_ARRAY[$index]} | sed 's/x64/x86_64/g;s/arm64/aarch64/g'`
+        echo -e "\nCopying ${PRODUCT} arch ${ARCHITECTURE_ARRAY[$index]} version ${VERSION} from ${TARBALL_ARRAY[$index]}"
+        cp -v ${TARBALL_ARRAY[$index]} ${DIR}/${PRODUCT}-${arch_uname}.tgz
+    done
+    ls -l $DIR
+fi
+
+# Build multi-arch images
```

Comment:
Can you convert all comments into log info (echo statements)? Will be easy to troubleshoot if the script if failing somewhere. Also it lets the user know the progress of the script

Reply:
No need, there is already outputs on each section for each command, which is redundant to add all again.

Reply:
"convert them" to echo statements for logging, Its easy to when you have logs. for debugging

Reply:
This is a good enhancement but not functionality breaking, and it has outputs from all existing commands that runs in the script. I would argue it already have all the things we need for debugging, even cp command it using -v param to print out.

Reply:
Agree with Sayali in general: human-readable log messages are better than echoing commands. In this case, the message says "Downloading ${PRODUCT} arch ${ARCHITECTURE_ARRAY[$index]} version ${VERSION} from ${URL}", which is human-readable enough.

Reply:
Since it is human-readable enough according to @camerski I will leave it as it is and we can improve later with another PR.